src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -449,7 +551,10 @@ private static dynamic ReAssignComplexDerivedType(dynamic originalValue, Type ne
                 {
                     //Now originalType = common base type of the derived complex types.
                     //OriginalNested Resource = T(of current Complex type). We are creating newOriginalNestedResource (T - new complex type).
-                    newOriginalNestedResource = Activator.CreateInstance(newType);
+                    if (TypeHelper.HasDefaultConstructor(newType))
+                    {
+                        newOriginalNestedResource = Activator.CreateInstance(newType);
+                    }
```

Comment:
I believe we should throw an exception if the type has no default constructor. Otherwise return a null will result into a `NullReferenceException` when we later call `CopyChangedValues`